app/services/external_payment_processor.rb

```diff
@@ -2,10 +2,25 @@ class ExternalPaymentProcessor
   require "#{Rails.root}/lib/stripe_payment_processor.rb"
 
   attr_accessor :processed, :error
-
+  
+  #creates a new instance of ExternalPaymentProcessor that sets either processed as the successful response or error as the error message
+  #based on whether the response from the payment processor returns an id.
   def self.create_payment_process(options={})
-    response = payment_processor.new(options).process_card
-    response == true ? new(processed: response) : new(error: response)
+    payment_response = payment_processor.new(options).process_card
+    create_new_instance_from_response(payment_response)
```

Comment:
Hi, I've refactored the code from the suggestions you made although the reason I did this abstraction was because both class methods were using exactly the same code and I was trying to DRY my code